src/main/java-templates/org/primefaces/component/picklist/PickListTemplate.java

```diff
@@ -76,34 +76,68 @@
     public Collection<String> getEventNames() {
         return EVENT_NAMES;
     }
+    
+    @Override
+    protected void validateValue(FacesContext facesContext, Object newValue) {
+        super.validateValue(facesContext, newValue);
+
+        DualListModel<?> newModel = (DualListModel<?>) newValue;
+        DualListModel<?> oldModel = (DualListModel<?>) this.getValue();
+        
+        String clientId = this.getClientId(facesContext);
+        String label = this.getLabel();
+        if (label == null) {
+            label = clientId;
+        }
+        
+        if (isRequired() && newModel.getTarget().isEmpty()) {
+            String requiredMessage = getRequiredMessage();
+            FacesMessage message = null;
 
-	protected void validateValue(FacesContext facesContext, Object newValue) {
-		super.validateValue(facesContext, newValue);
-		
-		DualListModel model = (DualListModel) newValue;
-		if(isRequired() && model.getTarget().isEmpty()) {
-			String requiredMessage = getRequiredMessage();
-			FacesMessage message = null;
-			
-			if(requiredMessage != null) {
-				message = new FacesMessage(FacesMessage.SEVERITY_ERROR, requiredMessage, requiredMessage);
+            if (requiredMessage != null) {
+                message = new FacesMessage(FacesMessage.SEVERITY_ERROR, requiredMessage, requiredMessage);
             }
             else {
-                String label = this.getLabel();
-                if(label == null) {
-                    label = this.getClientId(facesContext);
-                }
-
-	        	message = MessageFactory.getMessage(REQUIRED_MESSAGE_ID, FacesMessage.SEVERITY_ERROR, new Object[]{label});
-
+                message = MessageFactory.getMessage(REQUIRED_MESSAGE_ID, FacesMessage.SEVERITY_ERROR, new Object[] { label });
             }
+            facesContext.addMessage(clientId, message);
+            setValid(false);
+        }
+        
+        checkDisabled(facesContext, label, newModel.getSource(), oldModel.getSource());
+        checkDisabled(facesContext, label, newModel.getTarget(), oldModel.getTarget());
+    }
+	
+    /**
+     * Prohibits client-side manipulation of disabled entries, when CSS style-class ui-state-disabled is removed. See
+     * <a href="https://github.com/primefaces/primefaces/issues/2127">https://github.com/primefaces/primefaces/issues/2127</a>
+     *
+     * @param newEntries new/set entries of model source/target list
+     * @param oldEntries old/former entries of model source/target list
+     */
+    protected void checkDisabled(FacesContext facesContext, String label, List<?> newEntries, List<?> oldEntries) {
+        if (!isValid()) {
+            return;
+        }
+        
+        String varName = getVar();
+        String clientId = this.getClientId(facesContext);
+        for (int i = 0; i < newEntries.size(); i++) {
```

Comment:
Why not use the standard non-indexed iterator here, e.g. `for (final Object item : newEntries) {`